uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -250,25 +274,45 @@ define([
         vm.error = undefined;
       };
 
-      (function () {
-        // Create an empty leave request
-        vm.leaveRequest = LeaveRequestInstance.init({
-          contact_id: contactId //resolved from directive
-        }, false);
+      /**
+       * initializes the controller on loading the dialog
+       */
+      (function initController() {
+        if (directiveOptions.leaveRequest) {
+          //manager is responding to leave request
+          vm.uiOptions.isManager = directiveOptions.leaveRequest.contact_id != directiveOptions.contactId;
+          //owner is editing popup
+          vm.uiOptions.isOwner = directiveOptions.leaveRequest.contact_id == directiveOptions.contactId;
```

Comment:
Why are you not encapsulating this "check what role the current user has" logic in one single function? You have  it here, and then in `isManagedBy()`